packages/flutter_tools/lib/src/dart/package_map.dart

```diff
@@ -13,3 +22,32 @@ set globalPackagesPath(String value) {
 bool get isUsingCustomPackagesPath => _globalPackagesPath != null;
 
 String _globalPackagesPath;
+
+/// Load the package configuration from [file] or throws a [ToolExit]
+/// if the operation would fail.
+Future<PackageConfig> loadPackageConfigOrFail(File file, {
+  @required Logger logger,
+}) {
+  final FileSystem fileSystem = file.fileSystem;
+  return loadPackageConfigUri(
+    file.absolute.uri,
+    loader: (Uri uri) {
+      final File configFile = fileSystem.file(uri);
+      if (!configFile.existsSync()) {
+        return null;
+      }
+      return Future<Uint8List>.value(configFile.readAsBytesSync());
+    },
+    onError: (dynamic error) {
+      String message = '${file.path} does not exist.';
```

Comment:
`error` could be any exception from the `loader` callback. It would probably be good to at least get it into a trace message if not the error message.

Reply:
Added to trace